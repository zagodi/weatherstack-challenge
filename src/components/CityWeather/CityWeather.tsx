import React from "react";

import { useMemo } from "react";
import { Descriptions } from "antd";
import type { DescriptionsProps } from "antd";

import { CURRENT_WEATHER_RESPONSE } from "../../hooks/useGetCurrentWeather/useGetCurrentWeather";

import "./CityWeather.scss";

type PROPS = {
  dataSource: CURRENT_WEATHER_RESPONSE;
};

const CityWeather = ({ dataSource }: PROPS) => {
  const items: DescriptionsProps["items"] = useMemo(
    () => [
      {
        label: "City",
        children: dataSource.location.name,
        key: "name",
        labelStyle: { fontWeight: "bold" },
      },
      {
        label: "Country",
        children: dataSource.location.country,
        key: "age",
        labelStyle: { fontWeight: "bold" },
      },
      {
        label: "Temperature",
        children: dataSource.current.temperature,
        key: "temperature",
        labelStyle: { fontWeight: "bold" },
      },
      {
        label: "Weather Description",
        children: dataSource.current.weather_descriptions.join(", "),
        key: "weather_descriptions",
        labelStyle: { fontWeight: "bold" },
      },
      {
        label: "Wind Speed",
        children: dataSource.current.wind_speed,
        key: "wind_speed",
        labelStyle: { fontWeight: "bold" },
      },
      {
        label: "Humidity",
        children: dataSource.current.humidity,
        key: "humidity",
        labelStyle: { fontWeight: "bold" },
      },
      {
        label: "UV Index",
        children: dataSource.current.uv_index,
        key: "uv_index",
        labelStyle: { fontWeight: "bold" },
      },
      {
        label: "Precip",
        children: dataSource.current.precip,
        key: "precip",
        labelStyle: { fontWeight: "bold" },
      },
      {
        label: "Pressure",
        children: dataSource.current.pressure,
        key: "pressure",
        labelStyle: { fontWeight: "bold" },
      },
    ],
    []
  );

  return (
    <div className="city-weather-container">
      <Descriptions
        title="Weather Info"
        bordered
        layout="vertical"
        items={items}
      />
    </div>
  );
};

export default CityWeather;
