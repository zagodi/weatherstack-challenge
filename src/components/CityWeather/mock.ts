import { CURRENT_WEATHER_RESPONSE } from "../../hooks/useGetCurrentWeather/useGetCurrentWeather";

const mockDataSource: CURRENT_WEATHER_RESPONSE = {
  request: {
    type: "City",
    query: "New York, United States of America",
    language: "en",
    unit: "m",
  },
  location: {
    name: "New York",
    country: "United States of America",
    region: "New York",
    lat: "40.714",
    lon: "-74.006",
    timezone_id: "America/New_York",
    localtime: "2024-10-20 16:35",
    localtime_epoch: 1729442100,
    utc_offset: "-4.0",
  },
  current: {
    observation_time: "08:35 PM",
    temperature: 24,
    weather_code: 116,
    weather_icons: [
      "https://cdn.worldweatheronline.com/images/wsymbols01_png_64/wsymbol_0002_sunny_intervals.png",
    ],
    weather_descriptions: ["Partly cloudy"],
    wind_speed: 9,
    wind_degree: 230,
    wind_dir: "SW",
    pressure: 1024,
    precip: 0,
    humidity: 22,
    cloudcover: 25,
    feelslike: 24,
    uv_index: 1,
    visibility: 16,
  },
};

export default mockDataSource;
