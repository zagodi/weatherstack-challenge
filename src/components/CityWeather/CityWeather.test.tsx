import React from "react";

import { screen, render } from "@testing-library/react";

import CityWeather from "./CityWeather";
import mockDataSource from "./mock";

describe("<CityWeather />", () => {
  it("should render component correctly", () => {
    const { getByText, container } = render(
      <CityWeather dataSource={mockDataSource} />
    );

    expect(getByText("City")).toBeInTheDocument();
    expect(getByText("New York")).toBeInTheDocument();
    expect(screen.getAllByRole("cell")).toHaveLength(9);

    expect(container.firstChild).toMatchSnapshot();
  });
});
