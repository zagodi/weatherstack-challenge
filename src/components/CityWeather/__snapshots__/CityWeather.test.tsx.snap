// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CityWeather /> should render component correctly 1`] = `
<div
  class="city-weather-container"
>
  <div
    class="ant-descriptions ant-descriptions-bordered css-dev-only-do-not-override-1hpnbz2"
  >
    <div
      class="ant-descriptions-header"
    >
      <div
        class="ant-descriptions-title"
      >
        Weather Info
      </div>
    </div>
    <div
      class="ant-descriptions-view"
    >
      <table>
        <tbody>
          <tr
            class="ant-descriptions-row"
          >
            <th
              class="ant-descriptions-item-label"
              colspan="1"
            >
              <span
                style="font-weight: bold;"
              >
                City
              </span>
            </th>
            <th
              class="ant-descriptions-item-label"
              colspan="1"
            >
              <span
                style="font-weight: bold;"
              >
                Country
              </span>
            </th>
            <th
              class="ant-descriptions-item-label"
              colspan="1"
            >
              <span
                style="font-weight: bold;"
              >
                Temperature
              </span>
            </th>
          </tr>
          <tr
            class="ant-descriptions-row"
          >
            <td
              class="ant-descriptions-item-content"
              colspan="1"
            >
              <span>
                New York
              </span>
            </td>
            <td
              class="ant-descriptions-item-content"
              colspan="1"
            >
              <span>
                United States of America
              </span>
            </td>
            <td
              class="ant-descriptions-item-content"
              colspan="1"
            >
              <span>
                24
              </span>
            </td>
          </tr>
          <tr
            class="ant-descriptions-row"
          >
            <th
              class="ant-descriptions-item-label"
              colspan="1"
            >
              <span
                style="font-weight: bold;"
              >
                Weather Description
              </span>
            </th>
            <th
              class="ant-descriptions-item-label"
              colspan="1"
            >
              <span
                style="font-weight: bold;"
              >
                Wind Speed
              </span>
            </th>
            <th
              class="ant-descriptions-item-label"
              colspan="1"
            >
              <span
                style="font-weight: bold;"
              >
                Humidity
              </span>
            </th>
          </tr>
          <tr
            class="ant-descriptions-row"
          >
            <td
              class="ant-descriptions-item-content"
              colspan="1"
            >
              <span>
                Partly cloudy
              </span>
            </td>
            <td
              class="ant-descriptions-item-content"
              colspan="1"
            >
              <span>
                9
              </span>
            </td>
            <td
              class="ant-descriptions-item-content"
              colspan="1"
            >
              <span>
                22
              </span>
            </td>
          </tr>
          <tr
            class="ant-descriptions-row"
          >
            <th
              class="ant-descriptions-item-label"
              colspan="1"
            >
              <span
                style="font-weight: bold;"
              >
                UV Index
              </span>
            </th>
            <th
              class="ant-descriptions-item-label"
              colspan="1"
            >
              <span
                style="font-weight: bold;"
              >
                Precip
              </span>
            </th>
            <th
              class="ant-descriptions-item-label"
              colspan="1"
            >
              <span
                style="font-weight: bold;"
              >
                Pressure
              </span>
            </th>
          </tr>
          <tr
            class="ant-descriptions-row"
          >
            <td
              class="ant-descriptions-item-content"
              colspan="1"
            >
              <span>
                1
              </span>
            </td>
            <td
              class="ant-descriptions-item-content"
              colspan="1"
            >
              <span>
                0
              </span>
            </td>
            <td
              class="ant-descriptions-item-content"
              colspan="1"
            >
              <span>
                1024
              </span>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
